common/src/main/java/com/wynntils/handlers/chat/ChatHandler.java

```diff
@@ -197,9 +196,9 @@ private void processNewLines(LinkedList<Component> newLines) {
         LinkedList<Component> newChatLines = new LinkedList<>();
         LinkedList<Component> dialogue = new LinkedList<>();
 
-        CodedString firstLineCoded = ComponentUtils.getCoded(newLines.getFirst());
-        boolean isNpcConfirm = firstLineCoded.getMatcher(NPC_CONFIRM_PATTERN).find();
-        boolean isNpcSelect = firstLineCoded.getMatcher(NPC_SELECT_PATTERN).find();
+        StyledText firstLineCoded = StyledText.fromComponent(newLines.getFirst());
+        boolean isNpcConfirm = firstLineCoded.matches(NPC_CONFIRM_PATTERN);
+        boolean isNpcSelect = firstLineCoded.matches(NPC_SELECT_PATTERN);
```

Comment:
This is not equivalent. The old code uses `find`, and you use `matches` which greatly differs.

Reply:
there is a ^ and a $. Doesn't that mean, that it always matches the whole string?

Reply:
I didn't check the regexes for this info. However, please still use find (you can add StyledText methods) so it doesn't look confusing in the git history. If you are really bothered you can come back and change the regex etc.